internal/sandbox/wasi.go

```diff
@@ -360,3 +894,347 @@ func wasiROFlags(rflags int) (roflags wasi.ROFlags) {
 	}
 	return roflags
 }
+
+// wasiFS is an interface used to represent a sandbox file system.
+//
+// The interface has a single method allowing the sandbox to open the root
+// directory.
+type wasiFS interface {
+	PathOpen(ctx context.Context, lookupFlags wasi.LookupFlags, path string, openFlags wasi.OpenFlags, rightsBase, rightsInheriting wasi.Rights, fdFlags wasi.FDFlags) (anyFile, wasi.Errno)
+}
+
+// ThrottleFS wraps the file system passed as argument to apply the rate limits
+// r and w on read and write operations.
+//
+// The limits apply to all access to the underlying file system which may result
+// in I/O operations.
+//
+// Passing a nil rate limiter to r or w disables rate limiting on the
+// corresponding I/O operations.
+func ThrottleFS(f wasiFS, r, w *rate.Limiter) wasiFS {
```

Comment:
Seems odd to me that a package-public function signature uses private types. Should `wasiFS` be `WasiFS`?

Reply:
You're correct, this is temporary until I update the `wasiFS` implementations to use `FS` instead 👍